src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3054,6 +3067,21 @@ void addProperties(ImmutableArray<ParameterSymbol> recordParameters)
                         builder.InstanceInitializersForRecordDeclarationWithParameters.Insert(addedCount, new FieldOrPropertyInitializer.Builder(property.BackingField, paramList.Parameters[param.Ordinal]));
                         addedCount++;
                     }
+                    else if (existingMember is PropertySymbol { IsStatic: false, GetMethod: { } } prop
```

Comment:
Not blocking this PR:
@cston I understand that we only need to check the existence of the getter for purpose of synthesizing `Deconstruct`, but I wonder if we should more generally reject properties that are not `{get; init; }`.
It seems to me that having a `public int X { get; init; }` is part of the expectation/contract of  `record C(int X) { ... }` whether the property is synthesized or user-defined.

Reply:
Opened #45009.